src/TaskList.svelte

```diff
@@ -0,0 +1,87 @@
+<script>
+  import { filteredTasks, tasks } from "./stores";
+
+  import TaskListItem from "./TaskListItem.svelte";
+
+  export let alertsOn;
+
+  // function to mark a task as complete, or return it to the array
+  function completeTask(event) {
+    const taskToComplete = event.detail;
+
+    const compliments = [
+      "Nice going!",
+      "Nice job!",
+      "Look at you go!",
+      "You did it!",
+      "Take a bow!",
+      "Wow!",
+      "Woohoo!",
+      "Time for a Mai Tai! 🍹",
+    ];
+    const index = Math.floor(Math.random() * compliments.length);
+
+    taskToComplete.done = !taskToComplete.done;
+
+    if (taskToComplete.done) {
+      if (alertsOn) {
+        alert(compliments[index]);
+      }
+    }
+
+    // need to do this to refresh the styling
+    pushTaskChangeToStore(taskToComplete);
+  }
+
+  // function to remove the specified task from the array
+  function deleteTask(event) {
+    const taskToDelete = event.detail;
+    tasks.update((values) => values.filter((row) => row !== taskToDelete));
+  }
+
+  function editTask(event) {
+    const taskToEdit = event.detail;
+    pushTaskChangeToStore(taskToEdit);
+  }
+
+  /**
+   * Ensures the specified task is pushed to the store with any changes.
+   */
+  function pushTaskChangeToStore(updatedTask) {
+    tasks.update((values) =>
+      values.map((value) => (value === updatedTask ? updatedTask : value)),
+    );
+  }
+</script>
+
+<div>
+  <ul>
+    {#each $filteredTasks as task}
+      <TaskListItem
+        {task}
+        on:complete={completeTask}
+        on:delete={deleteTask}
+        on:edit={editTask}
+      />
+    {/each}
+  </ul>
+</div>
+
+<style>
```

Comment:
Notice how I'm not using CSS classes here. Since Svelte styles are scoped to the component, and there's only one `<div>` and `<ul>` element, you can just use those selectors directly without worrying about the styles getting overwritten.